lib/three_scale/middleware/presigned_downloads.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module Middleware
+    class PresignedDownloads
+      RE_SEPARATOR = ::File::SEPARATOR == "/" ? "/" : "[#{Regexp.escape(::File::SEPARATOR)}/]"
```

Comment:
Which separator other than `/` could be?

Reply:
I guess `\` for Windows users :grimacing: 

But I read somewhere that in Ruby the paths are "normalized", and `/` can be used always.

Also, in our `paperclip.yml` we have `path: ':storage_root/:url'`, so I guess Paperclip will use `/` in the path anyway?.. I might be wrong though.

Reply:
On windows could be `\` and `/`. Not sure in docker on windows, perhaps only on pure windows can be `\`. Still prefer to handle what Rack clean urls are using.

Reply:
@mayorova , it is not an issue how we generate the URLs. That's clearly just `/`.

The super frustrating part is looking at the URL and knowing whether it points at the files yo uwant to protect or from the others that can be accessed without a signed URL.

You can connect to the server with `nc` and craft any request you want, with any slashes and escape sequences.

So I try to err on the side of supporting whatever the `FileHandler` class is supporting although puma protects from a few (but not all) of the possible malicious paths. See the tests.

Reply:
But do we officially support running 3scale over Windows? I think we only support Openshift